src/test/java/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -1803,40 +1771,44 @@ public void throwLimitExceededException() {
         participantService.createParticipant(STUDY, PARTICIPANT, false);
     }
     
+    // This test does not make sense in that the caller is always operating on their own account.
+    // However adding involves adding a new substudy.
     @Test
-    public void updateIdentifiersFiltersSubstudyPermissions() {
-        // caller in question is in substudyB
-        BridgeUtils.setRequestContext(
-                new RequestContext.Builder().withCallerSubstudies(ImmutableSet.of("substudyA")).build());
+    public void updateIdentifiersCanAddExternalIdInOtherSubstudy() {
+        // caller in question is in substudyA
+        BridgeUtils.setRequestContext(new RequestContext.Builder()
+                .withCallerSubstudies(ImmutableSet.of(SUBSTUDY_ID)).build());
         
-        // account in question is in substudyA
-        mockHealthCodeAndAccountRetrieval();
-        when(externalIdService.getExternalId(TEST_STUDY, EXTERNAL_ID)).thenReturn(Optional.of(extId));
+        // account is associated to the EXTERNAL_ID in SUBSTUDY_ID
+        mockHealthCodeAndAccountRetrieval(EMAIL, null, EXTERNAL_ID);
+        
+        // extId is not in the same substudy. This should work
+        extId.setSubstudyId("anotherSubstudy");
+        extId.setIdentifier("newExtId");
+        when(externalIdService.getExternalId(TEST_STUDY, "newExtId")).thenReturn(Optional.of(extId));
         
-        account.setAccountSubstudies(ImmutableSet.of(AccountSubstudy.create(STUDY.getIdentifier(), "substudyB", ID)));
         when(accountDao.authenticate(STUDY, EMAIL_PASSWORD_SIGN_IN)).thenReturn(account);
         
-        try {
-            // And so this fails...
-            IdentifierUpdate update = new IdentifierUpdate(EMAIL_PASSWORD_SIGN_IN, null, null, EXTERNAL_ID);
-            participantService.updateIdentifiers(STUDY, CONTEXT, update);
-            fail("Should have thrown exception");
-        } catch(EntityNotFoundException e) {
-            assertEquals(e.getMessage(), "Account not found.");
-        }
-        verify(accountDao, never()).updateAccount(any(), any());
+        IdentifierUpdate update = new IdentifierUpdate(EMAIL_PASSWORD_SIGN_IN, null, null, "newExtId");
+        participantService.updateIdentifiers(STUDY, CONTEXT, update);
+        
+        verify(accountDao).updateAccount(accountCaptor.capture(), any());
```

Comment:
We should also verify that we call externalIdService.commitAssignExternalId()

Reply:
Added